File: day3/Quiz3/solution.py
def maze_runner(maze,list):
    result = "Lost" 
    row = -1
    col = 0
    for i in maze:
        row+=1
        if 2 in i :
            col = i.index(2)
            break
    
    for j in list:
        if j == "N" :
            row-=1
        elif j == "E":
            col+=1
        elif j == "W":
            col-=1
        else :
            row+=1
        
        if row < 0 or col < 0 or maze[row][col] == 1:
            result = "Dead"
            break
        elif  maze[row][col] == 3:
            result = "Finish"
            break
    
    print(result)




if __name__ == "__main__":
    maze = [[1,1,1,1,1,1,1], [1,0,0,0,0,0,3], [1,0,1,0,1,0,1], [0,0,1,0,0,0,1], [1,0,1,0,1,0,1], [1,0,0,0,0,0,1], [1,2,1,0,1,0,1]]
    load = ["N","N","N","N","N","E","E","E","E","E"]
    maze_runner(maze,load)

File: day3/Quiz3/solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;


public class MyClass {
    public static String maze_runner(int[][] maze , String[] list){
        String result = "Lost" ;
        int row = 0;
        int col = 0;
        int mazeLength = maze.length-1;

        for(row = 0; row <= mazeLength; row++  ){
            List<Integer> chkList = Arrays.stream(maze[row]).boxed().collect(Collectors.toList());
            if(chkList.indexOf(2) >= 0 ){
                col = chkList.indexOf(2);
                break;
            }
        }
        for(String flag:list){
            switch(flag){
                case "N" :
                    row = row-1;
                    break;
                case "E" :
                    col = col+1;
                    break;
                case "W" :
                    col = col-1;
                    break;
                case "S" :
                    row = row+1;
                    break;
            }
            if(row > mazeLength || row < 0 ||col > mazeLength || col < 0 || maze[row][col] == 1){
                result = "Dead";
                break;
            }else if(maze[row][col] == 3){
                result = "Finish";
                break;
            }
            
        }
            

        return result;
        
    }
    public static void main(String args[]) {     
        
        int[][] maze = {{1,1,1,1,1,1,1},
                        {1,0,0,0,0,0,3},
                        {1,0,1,0,1,0,1},
                        {0,0,1,0,0,0,1},
                        {1,0,1,0,1,0,1},
                        {1,0,0,0,0,0,1},
                        {1,2,1,0,1,0,1}};
        String[] load = {"N","N","N","N","N","E","E","S","S","S","S","S","S"};
        System.out.println(maze_runner(maze,load));
        
    }
}
